master/buildbot/steps/source/git.py

```diff
@@ -570,35 +567,32 @@ def applyAlready(res):
         d.addCallback(applyAlready)
         return d
 
+    @defer.inlineCallbacks
     def _sourcedirIsUpdatable(self):
         if self.slaveVersionIsOlderThan('listdir', '2.16'):
-            d = self.pathExists(self.build.path_module.join(self.workdir, '.git'))
+            git_path = self.build.path_module.join(self.workdir, '.git')
```

Comment:
Sometimes it helps, and sometimes it does not.

/me is not going to argue here :)